### ISSUE TITLE
Sled Agent: `PUT /omicron-zones` needs a way to specify TUF artifact for zone image

### ISSUE THREAD

**Issue by davepacheco** [oxidecomputer/omicron#7281]
The idea we've discussed is that `OmicronZoneConfig` would grow an `artifact_id` (or `artifact_sha256`?) field that identifies which artifact from the [TUF] Repo Depot should be used for this zone's root filesystem.

This is a little trickier than just migrating sled-agent-owned config to Nexus-owned config because the behavior is changing, too:

- if the property is not specified, then the image to be used is "whatever was put in the `install` dataset by the last mupdate"
- once the property is specified, then the image to be used is what's specified in the property

This likely depends on oxidecompute/omicron#4411.  If we _don't_ do that first: sled agent has no way to find the right control-plane artifact that contains the requested zone image.  (We could have the API identify the control plane artifact instead of the specific image within it.  But I think that's not what we want in the long term because it requires Sled Agent to know more about the structure of the TUF repo and it also requires the whole artifact to be present everywhere -- though that second part is already going to be true.)

**Comment by iliana** [comment]
Two questions come to mind:
- should sled-agent reject a zone config if it references an artifact that is not locally present?
- should sled-agent prevent itself from deleting repo artifacts that are referenced in the current zone config?

I think the answer to both is "yes"?

**Comment by davepacheco** [comment]
Yeah that makes sense to me (but both should be unexpected and should probably raise an alarm, I think -- which is probably just a `error!` level log message for now).
